### DISCUSSION TITLE
History not saved

### DISCUSSION THREAD

**Discussion by almereyda** [docmost/docmost#398] [Help / Questions] [answered]
When launching the Compose environment, no history is saved, despite the Websocket connection is working as expected.

This is differerent from #235.

**Comment by Philipinho** [comment, marked as answer]
The page history is saved 10 minutes from the last update.
Can you confirm if the page is older than 10 minutes?

**Reply by almereyda** [reply]
Thanks, I wasn't aware of the timer.

Indeed the pages were younger.

Does it seem feasible to also support

- manually saving versions (with description of the change)
- event-based detection of versions, e.g. when navigating out of a document, when a collaboration session is left, when a (collaboration) session didn't have new input in a few seconds / minutes.

I don't know how HedgeDoc or Outline detect versions, but they have some heuristic which also allows to grab revisions from young documents.

**Reply by Philipinho** [reply]
It’s first iteration was more like your second suggestion. I opted out to not bloat the database with unnecessary history changes.
 I’m open to further reviewing the process eventually. Thank you for your input.
